src/x86/intel_joule_expansion.c

```diff
@@ -149,6 +170,114 @@ mraa_joule_expansion_board()
     b->spi_bus[4].bus_id = 32765;
     b->spi_bus[4].slave_s = 2;
 
+    b->uart_dev_count = 2;
+    b->def_uart_dev = 0;
+    b->uart_dev[0].device_path = "/dev/ttyS0";
+    b->uart_dev[0].rx = 68;
+    b->uart_dev[0].tx = 7;
+    b->uart_dev[1].device_path = "/dev/ttyS1";
+    b->uart_dev[1].rx = 24;
+    b->uart_dev[1].tx = 22;
+
+    // Aio shield detection
+    // Only if i2c_designware.0 was successfully found
+    if (i2c_aio_bus != -1) {
+
+        syslog(LOG_NOTICE, "Attempting shield autodetection on I2C bus %d...", i2c_aio_bus);
+        mraa_i2c_context i2c = mraa_i2c_init_raw(i2c_aio_bus);
+
+        if (i2c == NULL)
+            syslog(LOG_ERR, "Failed to open I2C bus: %d", i2c_aio_bus);
+        else {
+            if (mraa_i2c_address(i2c, 0x49) != MRAA_SUCCESS)
+                syslog(LOG_ERR, "Failed to set I2C address: %d", 0x49);
+            if (mraa_i2c_read_word_data(i2c, 0x01) < 0) {
+                syslog(LOG_NOTICE, "No device at I2C address 0x49");
+
+                // no DFRobot shield, try Grove
+                if (mraa_i2c_address(i2c, 0x48) != MRAA_SUCCESS)
+                    syslog(LOG_ERR, "Failed to set I2C address: %d", 0x48);
+                if (mraa_i2c_read_word_data(i2c, 0x01) < 0)
+                    syslog(LOG_NOTICE, "No device at I2C address 0x48");
+                    // no shield
+
+                else {
+                    // load ads1015 at 0x48
+                    syslog(LOG_NOTICE, "Loading ti-ads1015 module for Grove Shield");
+                    if (system("modprobe ti-ads1015") < 0) {
```

Comment:
#770 submitted.

@Propanu I think it's pretty clear cut, let's document module usage and remove these. I also want to remove all the system() calls generally from this PR. To avoid loosing the history, maybe just create a new PR? That would likely be best :)